### planner environment settings ###############################################

# the size of the used hash map (the bigger the better); should be something
# with 2^X (initially 2^16=65536)
max_hash_size: 65536

# the step cost function that should be used to estimate the step costs of a step
# possible choices:
#  ConstStepCostEstimator: Each step has equal cost using primitives
#  EuclideanStepCostEstimator: Euclidean distance as cost using primitives
#  GprStepCostEstimator: Using learned GPR solution
#  MapStepCostEstimator: Using lookup map
#  BoundaryStepCostEstimator: Using boundary policy
#  DynamicsStepCostEstimator: Using boundary policy and respects dynamic behavior of robot
step_cost_estimator_type: BoundaryStepCostEstimator

# default cost to perform one footstep (in addition to ALL other costs, corresponds to 'meters')
const_step_cost_estimator:
  step_cost: 0.2

# parameters for BoundaryStepCostEstimator (multipliers)
boundary_step_cost_estimator:
  max_diff_z: 0.40                      # [m]
  long_step_dist: 0.30                  # [m]
  min_yaw_seperation_enlargement: 0.2   # [rad]
  yaw_enlarged_min_seperation: 0.23     # [m]
  cost_roll_abs: 0.5                    # [1/rad]
  cost_pitch_abs: 0.3                   # [1/rad]
  cost_yaw_rel: 1.0                     # [1/rad]
  cost_height_diff_rel: 0.5             # [1/m]

# parameters for DynamicsStepCostEstimator
dynamic_step_cost_estimator:
  lower_step_limit: 0.1
  upper_step_limit: 0.5
  max_near_distance: 1.0

# parameters for post processing step
post_processing:
  sway_duration_max_swing_dist: 0.5     # max swing distance (in direction the stand foot faces) before sway duration will be adjusted 
  sway_duration_min_normal_z: 0.95      # min value in z axis of normal before sway duration will be adjusted
  sway_duration_adjusted: 0.0           # adjusted sway duration
  toe_off_max_turn_rate: 0.4            # max turn rate [rad] before TOE-OFF will be disabled
  knee_nominal_max_step_down: 0.5       # max step down distance (positive) [m] before knee nominal will be adjusted
  knee_nominal_adjusted: 0.0            # adjusted knee nominal      

# the heuristic that should be used to estimate the step costs of a planning state
# possible choices: EuclideanHeuristic, EuclStepCostHeuristic, PathCostHeuristic
heuristic_type: EuclStepCostHeuristic

# cost to estimate the difference between the current orientation and the goal
# orientation; used in the heuristic it can easily overestimate the true path
# cost since this cost is not incorporated in each step
diff_angle_cost: 0.0

heuristic_scale: 1.0


### planner settings ###########################################################

# define the planner of choice provided by the SBPL package
# - ARAPlanner
# - ADPlanner
# - RSTARPlanner
planner_type: ARAPlanner

# search until a specific time limit is reached or first solution is found
search_until_first_solution: False

# the maximum search time if search_until_first_solution is set to False
allocated_time: 600.0

initial_epsilon: 8.0

forward_search: True

# the limit of changed states that decides whether to replan or to start a hole
# new planning task
changed_cells_limit: 20000

# if set to true, planner will shift goal pose if initially chosen is invalid (= occupied)
shift_goal_pose: True


### Planning parameterization for a large humanoid, similar to ASIMO or HRP-2 ##
# used for foot contact support collision check
foot_contact_support:
  min_sampling_steps_x: 6
  min_sampling_steps_y: 4
  max_sampling_steps_x: 32
  max_sampling_steps_y: 24
  max_intrusion_z: 0.0075
  max_ground_clearance: 0.01
  minimal_support: 0.80

accuracy:
  # collision check type (binary decoded)
  # - 1 feet
  # - 2 upper body
  # - 4 foot contact support
  collision_check_type: 3

  # footstep collision check accuracy
  # - 0 (just the circumcircle of the foot)
  # - 1 (just the incircle of the foot)
  # - 2 (circumcircle and incircle recursivly checked for the whole foot)
  collision_check_accuray: 2

  # discretization of continuous space (cell size in meters)
  cell_size: 0.025

  num_angle_bins: 32
